docs/topics/arrays.md

```diff
@@ -1,28 +1,57 @@
 [//]: # (title: Arrays)
 
-Arrays in Kotlin are represented by the `Array` class. It has `get()` and `set()` functions that turn into `[]` by operator overloading conventions,
-and the `size` property, along with other useful member functions:
+> Before you use an array, we strongly recommend that you consider using a [collection](collections-overview.md) instead.
+> Collections have many benefits:
+>   * They can be mutable or immutable
+>   * You can add or remove elements from mutable collections
+>   * They have no boxing overhead
+>
+{type="note"}
 
-```kotlin
-class Array<T> private constructor() {
-    val size: Int
-    operator fun get(index: Int): T
-    operator fun set(index: Int, value: T): Unit
+Arrays in Kotlin are represented by the `Array` class. Arrays are always mutable and have a fixed size.
```

Comment:
I actually don't think we mention anywhere on the page that when using the `Array` class, you're always storing _objects_. Again, this is what potentially influences boxing, and what you can use to contrast with the primitive arrays further down the page.